.github/workflows/test.yml

```diff
@@ -0,0 +1,102 @@
+name: Build/Lint/Test
+on:
+  push:
+    paths-ignore:
+      - 'README.md'
+      - 'CHANGELOG.md'
+  pull_request:
+    paths-ignore:
+      - 'README.md'
+      - 'CHANGELOG.md'
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+          cache: true
+
+      - name: Get dependencies
+        run: go mod download
+
+      - name: Build
+        run: make build
+
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'tools/go.mod'
+          cache: true
+          cache-dependency-path: tools/go.sum
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_wrapper: false
+
+      - name: Lint
+        run: make lint
+
+  test:
+    name: Matrix Acceptance Test
+    needs: build
+    runs-on: ubuntu-latest
+    services:
+      elasticsearch:
+        image: docker.elastic.co/elasticsearch/elasticsearch:${{ matrix.elasticsearch }}
+        env:
+          discovery.type: single-node
+          xpack.security.enabled: true
+          repositories.url.allowed_urls: https://example.com/*
+          path.repo: /tmp
+          ELASTIC_PASSWORD: password
```

Comment:
Might be worth moving this value to `test.env.ELASTIC_PASSWORD` so it can be shared with the `TF acceptance tests` step.

Reply:
Sounds good! Is that like this?
https://github.com/elastic/terraform-provider-elasticstack/pull/169/commits/42b38b36fe17a042e9f2ac505fad0ae6aad15ce0

Reply:
We also needed this
https://github.com/elastic/terraform-provider-elasticstack/pull/169/commits/066be9c139c151964441158305ea8c1e8191ac69